Ergofarg/Warhammer-Fantasy-6th-edition
# Fixing sacred spawnings

Sorry if there are errors with the changes as I'm very new to BattleScribe data file management.

I found in the current v1.20 release there was an issue where adding spawnings to a Saurus Warrior unit would make the unit disappear and sometimes contribute points to the special units. I've tried to simplify the selections to an inline selection group with a maximum of 2 selections. The modifier on the unit will set the category to Rare if there are more than 1 selection on this group and set it to Special if there is exactly 1 selection. The roster validation will throw an error if more than 2 spawnings are chosen as expected.

There is still a bug that I am unsure how to fix where the unit is added to the special or rare group and the totals are calculated however the unit doesn't appear until another change is triggered in the roster (eg adding a single model to another unit).

The spawnings for the Scar-Veteran are also recreated as an inline selection group so that the Carnosaur is no longer an option and the Cold One price is the correct 23 points (Lizardmen Army Book, page 61) if the spawning of Itzl is chosen.

While trying to learn how the conditions work I adjusted the Old One spawnings to be inline to simplify the interface for managing them (was a selection item that is added to the model). There is still a validation limit of 3 spawnings. I also implemented the constraint to make sure only a single model can have the Blessed Mark of the Old Ones (Lizardmen Army Book, page 51).

To test these items:
* Create new roster for Lizardmen - Standard
* Add an Old Blood and assign the Blessed Mark of the Old Ones, Itzl, Sotek and Chotek - this should result in a validation error
* Remove Chotek and the error disappears
* Expand Itzl and add a Carnosaur to confirm its still an option
* Add a Scar-Veteran with the Blessed Mark of the Old Ones, Itzl and Sotek - this should trigger an error for only a single Old One mark per roster and too many spawnings
* Remove the Old Ones and Sotek to clear the errors
* Expand Itzl and the label should no longer be Old One Mount
* Add 3 units of Saurus Warriors
* Add 4 spawnings to the last unit of warriors to confirm it is now Rare (it may be necessary to edit another unit to make it appear on the roster)
* Reduce the spawning selections to 2 to keep it as a rare choice
* Add 1 spawning to another core unit of warriors to confirm it is now a special unit (again, editing another unit may be needed)

I'm not sure why the Data Editor bumped the `gameSystemRevision` to `8` for this commit when the catalog version still says `6` in the data editor.

Please let me know how the PR looks and what needs tweaking.
